source/solvers/gd_navier_stokes.cc

```diff
@@ -766,17 +765,25 @@ GDNavierStokesSolver<dim>::set_initial_condition_fd(
   else if (initial_condition_type == Parameters::InitialConditionType::viscous)
     {
       this->set_nodal_values();
-      double viscosity =
-        this->simulation_parameters.physical_properties.fluids[0].viscosity;
-      this->simulation_parameters.physical_properties.fluids[0].viscosity =
-        this->simulation_parameters.initial_condition->viscosity;
+      std::shared_ptr<RheologicalModel> original_viscosity_model =
+        this->simulation_parameters.physical_properties_manager.get_rheology();
+
+      std::shared_ptr<Newtonian> temporary_rheology =
```

Comment:
maybe add a comment to explain what temporary_rheology is doing?

Reply:
Agreed :)!. Done